sklearn/ensemble/_gb.py

```diff
@@ -578,7 +578,16 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # The requirements of _decision_function (called in two lines
             # below) are more constrained than fit. It accepts only CSR
             # matrices.
-            X = check_array(X, dtype=DTYPE, order="C", accept_sparse="csr")
+            if self.warm_start:
+                X = check_array(
+                    X,
+                    dtype=DTYPE,
+                    order="C",
+                    accept_sparse="csr",
+                    force_all_finite=False,
+                )
+            else:
+                X = check_array(X, dtype=DTYPE, order="C", accept_sparse="csr")
```

Comment:
Hi @Geoffrey-Paris , thanks for your pull request... to bring more attention to your work please note that codecov is complaining about the lack of test coverage on new lines. Do you mind adding a test that would check the behavior of the condition? Thanks!